File: src/java/magmac/compile/StripRule.java
package magmac.compile;

import java.util.Optional;

public record StripRule(String leftKey, Rule rule, String rightKey) implements Rule {
    public StripRule(Rule rule) {
        this("", rule, "");
    }

    @Override
    public Optional<String> generate(MapNode node) {
        var leftString = node.findString(this.leftKey).orElse("");
        var rightString = node.findString(this.rightKey).orElse("");
        return this.rule.generate(node).map(generated -> leftString + generated + rightString);
    }

    @Override
    public Optional<MapNode> parse(String input) {
        return this.rule.parse(input.strip());
    }
}

File: src/main/windows/magma/compile/rule/StripRule.c
struct StripRule(String leftKey, Rule rule, String rightKey) implements Rule{};
Rule Strip_StripRule(String leftKey, Rule rule, String rightKey) implements Rule(Rule rule) {}
Rule Strip_StripRule(String leftKey, Rule rule, String rightKey) implements Rule(String left,  Rule rule,  String right) {}
CompileError> lex_StripRule(String leftKey, Rule rule, String rightKey) implements Rule(String content) {}
CompileError> generate_StripRule(String leftKey, Rule rule, String rightKey) implements Rule(Node node) {}
